src/metadata/trackmetadata.cpp

```diff
@@ -0,0 +1,121 @@
+#include "metadata/trackmetadata.h"
+
+#include "util/math.h"
+
+namespace Mixxx {
+
+/*static*/ const double TrackMetadata::BPM_UNDEFINED = 0.0;
+/*static*/ const double TrackMetadata::BPM_MIN = 0.0; // lower bound (inclusive)
+/*static*/ const double TrackMetadata::BPM_MAX = 300.0; // lower bound (inclusive)
+
+/*static*/ const float TrackMetadata::REPLAYGAIN_UNDEFINED = 0.0f;
+/*static*/ const float TrackMetadata::REPLAYGAIN_MIN = 0.0f; // lower bound (inclusive)
+/*static*/ const float TrackMetadata::REPLAYGAIN_0DB = 1.0f;
+
+namespace {
+
+double parseBpmString(const QString& sBpm, bool* pValid = 0) {
+    if (pValid) {
+        *pValid = false;
+    }
+    if (sBpm.trimmed().isEmpty()) {
+        return TrackMetadata::BPM_UNDEFINED;
+    }
+    bool bpmValid = false;
+    double bpm = sBpm.toDouble(&bpmValid);
+    if (bpmValid) {
+        if (TrackMetadata::BPM_UNDEFINED == bpm) {
+            // special case
+            if (pValid) {
+                *pValid = true;
+            }
+            return bpm;
+        }
+        while (TrackMetadata::BPM_MAX < bpm) {
+            // TODO(XXX): Why?
+            qDebug() << "Scaling BPM:" << bpm;
+            bpm /= 10.0;
+        }
+        if (TrackMetadata::isBpmValid(bpm)) {
+            if (pValid) {
+                *pValid = true;
+            }
+            return bpm;
+        } else {
+            qDebug() << "BPM out of range:" << bpm;
+        }
+    } else {
+        qDebug() << "Failed to parse BPM:" << sBpm;
+    }
+    return TrackMetadata::BPM_UNDEFINED;
+}
+
+float parseReplayGainDbString(QString sReplayGainDb, bool* pValid = 0) {
+    if (pValid) {
+        *pValid = false;
+    }
+    sReplayGainDb.remove("dB"); // TODO(XXX): Why?
```

Comment:
This should be 
remove ("dB", Qt::CaseInsensitive) 

Comment proposal

// format according to http://wiki.hydrogenaud.io/index.php?title=ReplayGain_1.0_specification#ID3v2:
// [-]a.bb dB 
// A robust player should be prepared to parse the following variations in either replay gain or peak level metadata:
// * Positive gains with leading '+'
// * More or fewer significant digits than specified in any field
// * Leading zeros or spaces in integer fields
// * Missing or malformed 'dB' suffix (e.g. no space between numeric digits and suffix, alternate capitalization)
// * Alternate capitalization of keys
// Other formatting errors indicate more severe problems and should result in player ignoring data as if the frame did not exist.